### PR TITLE
Work5

### PR DESCRIPTION
**Pull request by sp640919**
## Review

- [ ] Ensure your files are written following RST specs (*not MD!*)
- [x] Italian version
- [x] English version
- [ ] Example files 
- [ ] Ask for review

### CODE REVIEW

File: docs/en/metadata_aa.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      "oauth_authorization_server":{
+        ...
+      }
+      "oauth_resource":{
+        ...
+      }
+    }
+ }
+
+The AA Metadata from **"federation_entity"** MUST contain at least the following mandatory parameters:
```
Comment (peppelinux) [inline docs/en/metadata_aa.rst:26]: we can't use from, we can't traduce it literally, from has another meaning

File: docs/en/metadata_oidc_rp.rst [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      "federation_entity":{
+        ...
+      }
+      "openid_relying_party":{
+        ...
+      }
+    }
+ }
+
+The OP Metadata from **"federation_entity"** MUST contain at least the following mandatory parameters:
```
Comment (peppelinux) [inline docs/en/metadata_oidc_rp.rst:23]: typo: from

**Review by peppelinux — CHANGES_REQUESTED** [review]

File: docs/en/attributi_utente.rst [hunk 3]
```diff
@@ -0,0 +1,385 @@
+.. include:: ./common_definitions.rst
```
Comment (peppelinux) [inline docs/en/attributi_utente.rst:1]: TODO: tradurre

File: docs/en/authentication_response.rst [hunk 4]
```diff
@@ -0,0 +1,97 @@
+.. include:: ./common_definitions.rst
```
Comment (peppelinux) [inline docs/en/authentication_response.rst:1]: TODO: tradurre

File: docs/en/authorization_endpoint.rst [hunk 5]
```diff
@@ -0,0 +1,155 @@
+.. include:: ./common_definitions.rst
```
Comment (peppelinux) [inline docs/en/authorization_endpoint.rst:1]: TODO: tradurre

File: docs/en/avvisi_spid.rst [hunk 6]
```diff
@@ -0,0 +1,19 @@
+.. include:: ./common_definitions.rst
```
Comment (peppelinux) [inline docs/en/avvisi_spid.rst:1]: TODO: tradurre

File: docs/en/differenze_oidc_fed.rst [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+Client Registration
++++++++++++++++++++
+
+SPID supports only **automatic_client_registration**. The **implicit** mode is not supported.
+
+
+Listing endpoint
+++++++++++++++++
+
+In the SPID Federation, for this endpoint, the claim **entity_type** is added to all the `OIDC-FED`_ standard ones, in order to obtain a filter on the type of subordinate entities. This need apecifically allows filtering entities of type **federation_entity**, **openid_relying_party**, **openid_provider** and **oauth_resource**.
```
Comment (peppelinux) [inline docs/en/differenze_oidc_fed.rst:21]: standard ones?

File: docs/en/flusso_autenticazione.rst [hunk 8]
```diff
@@ -0,0 +1,52 @@
+.. include:: ./common_definitions.rst
```
Comment (peppelinux) [inline docs/en/flusso_autenticazione.rst:1]: TODO: tradurre

File: docs/en/introspection_endpoint.rst [hunk 9]
```diff
@@ -0,0 +1,139 @@
+.. include:: ./common_definitions.rst
```
Comment (peppelinux) [inline docs/en/introspection_endpoint.rst:1]: TODO: tradurre

File: docs/en/metadata_oidc_op.rst [hunk 10]
```diff
@@ -0,0 +1,155 @@
+.. include:: ./common_definitions.rst
+
+.. _MetadataOP:
+
+OpenID Connect Provider Metadata (OP)
++++++++++++++++++++++++++++++++++++++
+
+An OP MUST publish inside its EC a Metadata of type *federation_entity* and a Metadata of type *openid_provider*, as
```
Comment (peppelinux) [inline docs/en/metadata_oidc_op.rst:8]: inside its -> in its

File: docs/en/revocation_endpoint.rst [hunk 11]
```diff
@@ -0,0 +1,73 @@
+.. include:: ./common_definitions.rst
```
Comment (peppelinux) [inline docs/en/revocation_endpoint.rst:1]: TODO: tradurre

File: docs/en/token_endpoint.rst [hunk 12]
```diff
@@ -0,0 +1,268 @@
+.. include:: ./common_definitions.rst
```
Comment (peppelinux) [inline docs/en/token_endpoint.rst:1]: TODO: tradurre

File: docs/en/trust_negotiation.rst [hunk 13]
```diff
@@ -0,0 +1,121 @@
+.. include:: ./common_definitions.rst
+
+Trust Negotiation
+-----------------
+
+This section explains the ways of mutual recognition between RP and OP, the ways for the Leaves to recognize each other inside the same Federation and to obtain each other's Metadata.
```
Comment (peppelinux) [inline docs/en/trust_negotiation.rst:6]: Leaves -> Leafs

please correct all the occourrences in the texts

File: docs/en/userinfo_endpoint.rst [hunk 14]
```diff
@@ -0,0 +1,79 @@
+.. include:: ./common_definitions.rst
```
Comment (peppelinux) [inline docs/en/userinfo_endpoint.rst:1]: TODO: tradurre

**Review by peppelinux — CHANGES_REQUESTED** [review]
good work, let's continue with this groove @sp640919 !

File: docs/en/differenze_oidc_fed.rst (hunk 7 above)
Comment (sp640919) [reply docs/en/differenze_oidc_fed.rst:21]: ho messo "is added to all the other `OIDC-FED`_ standard claims"

File: docs/en/metadata_aa.rst [hunk 15]
```diff
@@ -0,0 +1,101 @@
+.. include:: ./common_definitions.rst
+
+
+Attribute Authority Metadata
+++++++++++++++++++++++++++++
+
+An AA MUST publish in its EC a Metadata with *federation_entity*, a Metadata with *oauth_authorization_server*
```
Comment (peppelinux) [inline docs/en/metadata_aa.rst:7]: rephrase and also align the it version please

An AA MUST publish in its EC a *federation_entity* Metadata and an *oauth_resource* Metadata, if the resources are protected it MUST also publish an *oauth_authorization_server* Metadata.

File: docs/en/revocation_endpoint.rst [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+----------------------------
+
+An RP CAN request the revocation of an Access Token or a Refresh Token issued by an OP.
+
+When the user logs out, the RP MUST revoke the Access Token in its possession.
+
+.. note::
+  The Access Token revocation implies revoking all the Refresh Tokens linked to it.
+
+The OP WILL HAVE TO revoke the token specified in the request and WILL HAVE TO end the Single Sign-On session, if it is still active.
```
Comment (peppelinux) [inline docs/en/revocation_endpoint.rst:13]: Do we have WILL HAVE TO in the normative lagnuage section? If not please change with a defined normative language

File: docs/en/seccons_bcps.rst [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+-----------------------
+
+In this section we describe some security considerations in the OIDC Federation scope.
+
+
+Trust Marks as deterrent against abuses
++++++++++++++++++++++++++++++++++++++++
+
+The TM implementation and the filter on the TMs in the phase of Metadata Discovery, turn out to be necessary
+against attacks aimed at the resource consumption. An OP that is attacked at its *authorization* endpoint, with an high number of connections containing fake **client_id** and **authority_hints**, would produce
```
Comment (peppelinux) [inline docs/en/seccons_bcps.rst:15]: please rephrase "n OP that is attacked at its"

it doesn't sound good

File: docs/en/token_endpoint.rst [hunk 18]
```diff
@@ -0,0 +1,273 @@
...
+   :header-rows: 1
+
+   * - **Claim**
+     - **Description**
+     - **Supported by**
+   * - **access_token**
+     - The Access Token, in signed JWT format, allows accessing the UserInfo Endpoint for obtaining the attributes.
+     - |spid-icon| |cieid-icon|
+   * - **token_type**
+     - Type of *Access Token* returned. It must always be equal to **Bearer**
```
Comment (peppelinux) [inline docs/en/token_endpoint.rst:141]: missing period char at the end, please check also the followings

File: docs/en/token_endpoint.rst [hunk 19]
```diff
@@ -0,0 +1,273 @@
...
+   * - **exp**
+     - UNIX Timestamp with the expiry time of the JWT, coded as NumericDate as indicated in :rfc:`7519`
+     - |spid-icon| |cieid-icon|
+   * - **jti** 
+     - String. Unique identifier of the ID Token that the client can use for preventing a reuse, refusing the
+       ID Token if it has already been processed. It must be difficult to guess by an attacker and composed of a casual string. It is suggested to use the *uuid4* generator.
+     - |spid-icon| |cieid-icon|
+   * - **nonce** 
+     - String. Casual string generated by the RP for each user's session and sent in the 
+       Authentication Request (claim nonce), aimed at mitigating replay attacks. The client must make sure that it has the same value as the one tha has been sent in the Authentication Request.
```
Comment (peppelinux) [inline docs/en/token_endpoint.rst:226]: aimed at -> aimed to or amined for the mitigation of ...

File: docs/en/trust_marks.rst [hunk 20]
```diff
@@ -0,0 +1,245 @@
...
+             "https://public.intermediary.spid.it/"
+         ],
+         "https://registry.agid.gov.it/openid_relying_party/private/":[
+             "https://registry.spid.agid.gov.it/",
+             "https://private.other.intermediary.it/"
+         ]
+     }
+ }
+
+Each member entity MUST expose in its configuration (EC), the TMs that have been released by the issuing authority.
```
Comment (peppelinux) [inline docs/en/trust_marks.rst:46]: Entity, uppercase

File: docs/en/trust_marks.rst [hunk 21]
```diff
@@ -0,0 +1,245 @@
...
+Attribute Authorities (AA).
+
+The TA defines the subjects that are enabled to issue TMs that are recognizable inside the Federation,
+and it does it with the claim **trust_marks_issuers**, that is present in its Entity Configuration. 
+The value of the attribute **trust_marks_issuers** is composed by a JSON Object whose keys are the TM identifiers and whose values are the list of the identifiers (URLs) of the entities that are enabled 
+to issue them.
+
+The Trust Marks represent the first filter for establishing the trust among the parties. 
+They are essential elements for starting the Metadata resolution. 
+In their absence, an entity is not recognized as a member inside the Federation.
```
Comment (peppelinux) [inline docs/en/trust_marks.rst:58]: entity -> Entity

File: docs/en/trust_marks.rst [hunk 22]
```diff
@@ -0,0 +1,245 @@
...
+      - UNIX Timestamp with the JWT generation time, coded as NumericDate as indicated at :rfc:`7519`
+      - |spid-icon| |cieid-icon|
+    * - **logo_uri**
+      - String. An URL that points to the logo that represents the Trust Mark.
+      - |spid-icon| |cieid-icon|
+    * - **exp**
+      - UNIX Timestamp with the JWT expiry time, coded as NumericDate as indicated at :rfc:`7519`
+      - |spid-icon| |cieid-icon|
+    * - **ref**
+      - String. URL that points to information present on the web, about this Trust Mark
```
Comment (peppelinux) [inline docs/en/trust_marks.rst:212]: information present on the web 

please rephrase

File: docs/en/trust_marks.rst [hunk 23]
```diff
@@ -0,0 +1,245 @@
...
+      - |spid-icon| |cieid-icon|
+    * - **organization_type**
+      - String. Specifies if the entity belongs to the Italian Public Administration or the private sector (**public** or **private**)
+      - |spid-icon| |cieid-icon|
+    * - **id_code**
+      - String. Identification code of the Organization. Depending on the organization type, it must be
```
Comment (peppelinux) [inline docs/en/trust_marks.rst:218]: Organization is uppercased or not? IS this a defined term? if yes, uppercase the next def

File: docs/en/trust_marks_spid.rst [hunk 24]
```diff
@@ -0,0 +1,95 @@
...
+
+
+Trust Marks for SPID
+++++++++++++++++++++
+
+
+Examples of Trust Marks for SPID
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The TMs issued for the Leaves MUST be published by the Leaves themselves in their own **Entity Configuration**,
```
Comment (peppelinux) [inline docs/en/trust_marks_spid.rst:11]: Leaves -> Leafs

please do a global check for this typo

File: docs/en/trust_marks_spid.rst [hunk 25]
```diff
@@ -0,0 +1,95 @@
...
+inside the claim **trust_marks**. This is composed by lists of JSOB objects, each of them MUST contain at least
```
Comment (peppelinux) [inline docs/en/trust_marks_spid.rst:12]: typo: JSOB

File: docs/en/trust_negotiation.rst (hunk 13 above)
Comment (peppelinux) [inline docs/en/trust_negotiation.rst:6]: LEaves, again

File: docs/en/trust_negotiation.rst [hunk 26]
```diff
@@ -0,0 +1,121 @@
...
+
+
+Relying Party
++++++++++++++
+
+The RP gets the list of the OPs in JSON format by querying the :ref:`endpoint list<federation_endpoint>`, 
+available by the :ref:`Trust Anchor<Esempio_EN3>`. For each subject contained in the :ref:`answer<Esempio_EN3.1>` and corresponding to an OP, the RP :ref:`requires<Esempio_EN2>` and obtains the self-signed Entity Configuration by the OP.
+
+For each EC of the OPs, the RP tests the content signature by using the public key obtained by the Entity
```
Comment (peppelinux) [inline docs/en/trust_negotiation.rst:15]: "tests the content signature" -> validates the signature

File: docs/en/trust_negotiation.rst [hunk 27]
```diff
@@ -0,0 +1,121 @@
...
+Statement released by the Trust Anchor. After testing the Entity Configuration signature with the TA's public
```
Comment (peppelinux) [inline docs/en/trust_negotiation.rst:16]: testing -> validating

File: docs/en/trust_negotiation.rst [hunk 28]
```diff
@@ -0,0 +1,121 @@
...
+   Entity Configuration URL is |br|
+   ``https://rp.servizi-spid.it/oidc/.well-known/oidc-federation``.
+
+If not present at the end of the URL that identifies a subject, the slash mark ("/") must be added before
+concatenating the web path to the .well-known resource.
+
+Once the RP is recognized as part in the Federation, it gets the permission to make an Authentication Request.
+An OP that doesn't recognize the RP that makes the request, can correctly resolve the Trust. The OP starts
+requesting the Entity Configuration of the RP at the .well-known endpoint of the RP and, following the path
+provided by the *authority_hint*, reaches the Trust root, that's the TA. At each chain step, the OP can perform all the security controls by requesting the Entity Statements to each entity and validating the Trust Marks and the signatures. The following picture is a representative example of how the Trust Chain works.
```
Comment (peppelinux) [inline docs/en/trust_negotiation.rst:112]: "the Trust root, that's the TA" -> the TA

**Review by agcolella — APPROVED** [review]

File: docs/en/seccons_bcps.rst [hunk 29]
```diff
@@ -0,0 +1,71 @@
...
+
+The threshold **max_path_lenght** is applied to the vertical navigation and, after exceeding this threshold without finding a TA, the procedure of Metadata Discovery MUST be stopped. Consider the example of an RP that's a subordinate of an SA that's in turn a subordinate of another SA, while the **max_path_lenght** claim is equal to 1 and, after exceeding this threshold without finding the Trust Anchor, the procedure MUST be stopped.
+
+At the same time, the specifications of OIDC Federation 1.0 don't define a limit of the number of **authority_hints**, and this is because no TA can limit the number of Federations in which a member can take part. For this reason, it is useful that implementers adopt a maximum limit to the number of elements allowed inside the array authority_hints. The reason is avoiding that an exaggerated number of URLs contained in the list **authority_hints**, due to a bad configuration of a Leaf, produce an excessive resource consumption.
```
[the remark this answers is not in the document]
Comment (sp640919) [reply docs/en/seccons_bcps.rst:32]: specifications if the (plural) subject -> don't

File: docs/en/soggetti_aggregatori.rst [hunk 30]
```diff
@@ -0,0 +1,41 @@
+.. include:: ./common_definitions.rst
+
+.. _Soggetti_aggregatori:
+
+Intermediate Entities
+^^^^^^^^^^^^^^^^^^^^^
+
+An Intermediate Entity (SA) can register pre-existing RPs that are compliant to the OIDC-FED standard and lead to domains that are external to theirs, or that mask their subordinates behind them. In the first case, the SA is of type *Transparent* (**Light Aggregator**), and in the second case it is of type *Proxy* (**Full Aggregator**)
+
+The **Light** SAs register pre-existing RPs that are OIDC-FED-compliant and publish the ESs referred to them.
```
[the remark this answers is not in the document]
Comment (sp640919) [reply docs/en/soggetti_aggregatori.rst:10]: Now I have changed it to "compliant to OIDC-FED", that stands for the Italian "conformi a OIDC-FED"

File: docs/en/soggetti_aggregatori.rst [hunk 31]
```diff
@@ -0,0 +1,41 @@
...
+
+The **Full** SAs arrange building an authentication and federation interface, on behalf of their own subordinates, by using web resources that are usually exposed inside their own domain. This kind of SAs expose for each of their subordinates, the following resources:
+
+    - **.well-known/openid-federation**, containing a *subject identifier* or the unique RP;
+    - Authorization callback endpoint for obtaining the auth code by the OP (**redirect_uri**).
+
+The **Full** type SAs MUST add the **id_code** (as defined in the Section :ref:`Trust Mark Composition <ComposizioneTM>`) in the web path inside the client_id that identifies the subordinate entity ``<SA_dominio>/<id_code>/``.
```
[the remark this answers is not in the document]
Comment (sp640919) [reply docs/en/soggetti_aggregatori.rst:17]: The Italian source was: "all’interno del webpath all’interno del client_id che identifica l’aggregato ``<SA_dominio>/<id_code>/``"
So I have changed the English version to this: "inside the web path, which in turn is inside the client_id that identifies the subordinate entity ``<SA_domain>/<id_code>/``.", that is all what I have understood. If it's still not clear, please help with the Italian original

File: docs/en/token_endpoint.rst (hunk 19 above)
Comment (sp640919) [reply docs/en/token_endpoint.rst:226]: aimed regge la preposizione "at"

File: docs/en/trust_marks_spid.rst (hunk 24 above)
Comment (sp640919) [reply docs/en/trust_marks_spid.rst:11]: No. The plural of Leaf is "Leaves"

**Pull request merged**